backend/internal/role/service.go

```diff
@@ -0,0 +1,614 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// Package role provides role management functionality.
+package role
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/asgardeo/thunder/internal/group"
+	oupkg "github.com/asgardeo/thunder/internal/ou"
+	serverconst "github.com/asgardeo/thunder/internal/system/constants"
+	"github.com/asgardeo/thunder/internal/system/error/serviceerror"
+	"github.com/asgardeo/thunder/internal/system/log"
+	"github.com/asgardeo/thunder/internal/system/utils"
+	"github.com/asgardeo/thunder/internal/user"
+)
+
+const loggerComponentName = "RoleMgtService"
+
+// RoleServiceInterface defines the interface for the role service.
+type RoleServiceInterface interface {
+	GetRoleList(limit, offset int) (*RoleListResponse, *serviceerror.ServiceError)
+	CreateRole(request CreateRoleRequest) (*Role, *serviceerror.ServiceError)
```

Comment:
I'd prefer to define separate structs for service layer rather than using request models.
WDYT?